src/avif.c

```diff
@@ -208,21 +245,8 @@ avifResult avifImageCopy(avifImage * dstImage, const avifImage * srcImage, avifP
             return allocationResult;
         }
     }
-    for (int plane = AVIF_CHAN_Y; plane <= AVIF_CHAN_A; ++plane) {
-        uint8_t * dstRow = avifImagePlane(dstImage, plane);
-        if (!dstRow) {
-            continue;
-        }
-        const uint8_t * srcRow = avifImagePlane(srcImage, plane);
-        uint32_t srcRowBytes = avifImagePlaneRowBytes(srcImage, plane);
-        uint32_t dstRowBytes = avifImagePlaneRowBytes(dstImage, plane);
-        uint32_t planeWidthBytes = avifImagePlaneWidth(dstImage, plane) << (dstImage->depth > 8);
-        uint32_t planeHeight = avifImagePlaneHeight(dstImage, plane);
-        for (uint32_t j = 0; j < planeHeight; ++j) {
-            memcpy(dstRow, srcRow, planeWidthBytes);
-            srcRow += srcRowBytes;
-            dstRow += dstRowBytes;
-        }
+    if (avifImageCopySamples(dstImage, srcImage, planes) != AVIF_RESULT_OK) {
+        assert(AVIF_FALSE);
```

Comment:
Return AVIF_FALSE. Assertions are compiled away in release builds.

Note: If `avifImageCopySamples()` is only used in places where it is obvious that `dstImage` and `srcImage` have the same properties, we can change `avifImageCopySamples()` to return void and change its checks to assertions.